src/Analyzers/CSharp/Tests/NamingStyles/NamingStylesTests.cs

```diff
@@ -1334,5 +1334,36 @@ public async Task TestRecordParameter_ParameterFormattedAsProperties()
 @"public record Foo(int [|MyInt|]);",
                 options: s_options.MergeStyles(s_options.PropertyNamesArePascalCase, s_options.ParameterNamesAreCamelCaseWithPUnderscorePrefix));
         }
+
+        [Theory]
+        [InlineData("_")]
+        [InlineData("_1")]
+        [InlineData("m_1")]
```

Comment:
i don't really ge the `m_1` case.  whyis this treated the same as `_` and `_1`?

Reply:
https://github.com/huoyaoyuan/roslyn/blob/4239a7ae9231e4df5c96efbae7413d3807eb6aeb/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/NamingStyle.cs#L376
The logic strips `m_` like `_`.

Reply:
hrmm... there's a strong difference though between these.  i think we're classifying `_` and `_0` (_1, _2, _3, etc.) as discards.  we wouldn't do the same with `m_1`.

Reply:
I think it would need more design about how to determine naming style when there are mixed type characters.
It falls into a similar situation I've met in dotnet/runtime#42009

Reply:
i think the design is pretty simple here.  Treat underscore-followed-by-number as a discard :)  I would limit it to just that.

Reply:
+1 to what @CyrusNajmabadi said.

Reply:
Current code fix renames `m_1` to `1`. Is there any consideration to change the logic?

Reply:
> Current code fix renames m_1 to 1. Is there any consideration to change the logic?

Yes.  That is clearly wrong.  But it woudl also be out of scope of this PR IMO :)